internal/server/singleprocess/poll_test.go

```diff
@@ -75,3 +75,195 @@ func TestServicePollQueue(t *testing.T) {
 	require.NoError(err)
 	require.Equal(len(raw), len(raw2))
 }
+
+func TestProjectPollHandler(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+
+	// Create our server
+	impl, err := New(WithDB(testDB(t)))
+	require.NoError(err)
+	client := server.TestServer(t, impl)
+
+	// Create a project
+	_, err = client.UpsertProject(ctx, &pb.UpsertProjectRequest{
+		Project: serverptypes.TestProject(t, &pb.Project{
+			Name: "Example",
+			DataSource: &pb.Job_DataSource{
+				Source: &pb.Job_DataSource_Local{
+					Local: &pb.Job_Local{},
+				},
+			},
+			DataSourcePoll: &pb.Project_Poll{
+				Enabled:  true,
+				Interval: "15ms",
+			},
+			Applications: []*pb.Application{
+				{
+					Project: &pb.Ref_Project{Project: "Example"},
+					Name:    "apple-app",
+					StatusReportPoll: &pb.Application_Poll{
+						Enabled: false,
+					},
+				},
+			},
+		}),
+	})
+	require.NoError(err)
+
+	// Grab next poll time
+	state := testServiceImpl(impl).state
+	p, pollTime, err := state.ProjectPollPeek(nil)
+	require.NoError(err)
+	require.NotNil(p)
+	require.NotNil(pollTime)
+
+	// Wait a bit. The interval is so low that this should trigger
+	// multiple loops through the poller. But we want to ensure we
+	// have only one poll job queued.
+	time.Sleep(50 * time.Millisecond)
+
+	// Check for our condition, we do eventually here because if we're
+	// in a slow environment then this may still be empty.
+	require.Eventually(func() bool {
+		// We should have a single poll job
+		var jobs []*pb.Job
+		raw, err := testServiceImpl(impl).state.JobList()
+		for _, j := range raw {
+			if j.State != pb.Job_ERROR {
+				jobs = append(jobs, j)
+			}
+		}
+
+		if err != nil {
+			t.Logf("err: %s", err)
+			return false
+		}
+
+		return len(jobs) == 1
+	}, 5*time.Second, 50*time.Millisecond)
+
+	// Cancel our poller to ensure it stops
+	testServiceImpl(impl).Close()
+
+	// ensure the next poll is after the initial poll before waiting
+	// next poll time gets set when a project poll is marked complete
+	p, nextPollTime, err := state.ProjectPollPeek(nil)
+	require.NoError(err)
+	require.NotNil(p)
+	require.NotNil(nextPollTime)
+	require.True(nextPollTime.After(pollTime))
+}
+
+func TestApplicationPollHandler(t *testing.T) {
```

Comment:
Big picture comment 💭 : these tests are _fine_ and good! But now that we extracted poll handling to an interface, we can probably test the interface methods do what we expect.

For example: these big picture tests should test that the poller handles the interface properly (and probably use a mock/test interface to test all the edges). Separately, we should test that our interface implementation _returns_ the proper data for each method call.

The actionable feedback here would be to make those lines clear. I would say to probably just skip that in this PR and noodle on it for another PR.